packages/devtools/lib/src/memory/memory.dart

```diff
@@ -185,6 +228,205 @@ class MemoryScreen extends Screen with SetStateMixin {
+    instanceTable.element.add(spinner);
+
+    // There's an instances table up.
+    final List<InstanceSummary> instances = instanceTable.data;
+    int row = 0;
+    for (InstanceSummary instance in instances) {
+      // Check the field in each instance looking to find the object being held
+      // (the hashCode passed in matches the particular field's hashCode)
+      final evalResult = await evaluate(instance.objectRef, '$field.hashCode');
+      if (evalResult != null && evalResult.valueAsString == hashCode) {
+        // Found the object select the instance.
+        instanceTable.selectByIndex(row, scrollBehavior: 'auto');
+        spinner.element.remove();
+        return row;
+      }
+      row++;
+    }
+
+    spinner.element.remove();
+
+    framework.toast('Unable to find instance for field $field [$hashCode]',
+        title: 'Error');
+
+    return -1;
+  }
+
+  void _resetHistory() {
+    history..clazz('history-navigation-hidden', removeOthers: true);
+    history.clear();
+    memoryPath = NavigationPath();
+  }
+
+  /// Finish callback from search class selected (auto-complete).
+  void _callbackPopupSelectClass([bool cancel]) {
+    if (cancel) {
+      heapPopup.matcher.reset();
+      heapPopupList.reset();
+    } else {
+      // Reset memory history selecting a class.
+      _resetHistory();
+
+      // Highlighted class is the class to select.
+      final String classSelected = heapPopupList.highlightedItem;
+      _selectClass(classSelected);
+    }
+
+    // Done with the popup.
+    heapPopup.hide();
+  }
+
+  Future<void> _selectInstanceByHashCode(String objectHashCode) async {
+    // There's an instances table up.
+    final Table<Object> instanceTable = tableStack.last;
+    final List<InstanceSummary> instances = instanceTable.data;
+    int row = 0;
+    for (InstanceSummary instance in instances) {
+      // Check each instance looking to find a particular object.
+      final evalResult = await evaluate(instance.objectRef, 'hashCode');
+      if (evalResult != null && evalResult.valueAsString == objectHashCode) {
+        // Found the object select the instance.
+        instanceTable.selectByIndex(row, scrollBehavior: 'auto');
+        return;
+      }
+      row++;
+    }
+
+    framework.toast('Unable to find instance [$objectHashCode]',
+        title: 'Error');
+  }
+
+  void selectClassInstance(String className, String objectHashCode) {
+    // Remove selection in class list.
+    tableStack.first.clearSelection();
+    Timer.periodic(const Duration(milliseconds: 100), (Timer timer) {
+      if (!tableStack.first.hasSelection) {
+        // Wait until the class list has no selection.
+        timer.cancel();
+      }
+    });
+
+    // Select the class (don't record this select in memory history). The
+    // memoryPath will be added by NavigationState.inboundSelect - see below.
+    _selectClass(className, false);
+
+    Timer.periodic(const Duration(milliseconds: 100), (Timer timer) async {
```

Comment:
same question here: can we await a completer instead?

Reply:
Need a Table event - look at adding.